specification/search/data-plane/Azure.Search/stable/2024-07-01/searchindex.json

```diff
@@ -0,0 +1,2111 @@
+          },
+          {
+            "value": "full",
+            "name": "Full",
+            "description": "Uses the full Lucene query syntax for searches. Search text is interpreted using the Lucene query language which allows field-specific and weighted searches, as well as other advanced features."
+          },
+          {
+            "value": "semantic",
+            "name": "Semantic",
+            "description": "Best suited for queries expressed in natural language as opposed to keywords. Improves precision of search results by re-ranking the top search results using a ranking model trained on the Web corpus."
+          }
+        ]
+      },
+      "description": "Specifies the syntax of the search query. The default is 'simple'. Use 'full' if your query uses the Lucene query syntax."
+    },
+    "VectorQuery": {
+      "type": "object",
+      "discriminator": "kind",
+      "properties": {
+        "kind": {
+          "$ref": "#/definitions/VectorQueryKind",
+          "description": "The kind of vector query being performed.",
+          "x-nullable": false
+        },
+        "k": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Number of nearest neighbors to return as top hits."
+        },
+        "fields": {
+          "type": "string",
+          "description": "Vector Fields of type Collection(Edm.Single) to be included in the vector searched."
+        },
+        "exhaustive": {
+          "type": "boolean",
+          "description": "When true, triggers an exhaustive k-nearest neighbor search across all vectors within the vector index. Useful for scenarios where exact matches are critical, such as determining ground truth values."
+        },
+        "oversampling": {
+          "type": "number",
+          "format": "double",
+          "description": "Oversampling factor. Minimum value is 1. It overrides the 'defaultOversampling' parameter configured in the index definition. It can be set only when 'rerankWithOriginalVectors' is true. This parameter is only permitted when a compression method is used on the underlying vector field."
+        },
+        "weight": {
+          "type": "number",
+          "format": "float",
+          "description": "Relative weight of the vector query when compared to other vector query and/or the text query within the same search request. This value is used when combining the results of multiple ranking lists produced by the different vector queries and/or the results retrieved through the text query. The higher the weight, the higher the documents that matched that query will be in the final ranking. Default is 1.0 and the value needs to be a positive number larger than zero. "
+        }
+      },
+      "required": [
+        "kind"
+      ],
+      "description": "The query parameters for vector and hybrid search queries."
+    },
+    "RawVectorQuery": {
```

Comment:
Could `"x-ms-client-name": "VectorizedQuery"` be added to this, all the SDKs used this rename and this would allow us to remove a client-side Swagger transformation.

Reply:
added :)